### PR TITLE
Fix naming for base64 functions

### PR DESCRIPTION
**Pull request by czentgr**

### CODE REVIEW

**Comment by czentgr** [comment]
Fuzzer test result (ran 60s default test)
```
I0324 09:47:02.292660 407313 ExpressionFuzzer.cpp:1158] ==============================> Done with iteration 14863
I0324 09:47:02.292754 407313 ExpressionFuzzer.cpp:972] ==============================> Top 10 by number of rows processed
I0324 09:47:02.292757 407313 ExpressionFuzzer.cpp:974] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I0324 09:47:02.292760 407313 ExpressionFuzzer.cpp:978] cast 3162 5.98% 119927
I0324 09:47:02.292801 407313 ExpressionFuzzer.cpp:978] multiply 1686 3.19% 30984
I0324 09:47:02.292817 407313 ExpressionFuzzer.cpp:978] minus 1712 3.24% 30365
I0324 09:47:02.292821 407313 ExpressionFuzzer.cpp:978] plus 1655 3.13% 28894
I0324 09:47:02.292825 407313 ExpressionFuzzer.cpp:978] sign 1626 3.08% 28660
I0324 09:47:02.292829 407313 ExpressionFuzzer.cpp:978] abs 1688 3.19% 27995
I0324 09:47:02.292832 407313 ExpressionFuzzer.cpp:978] mod 1671 3.16% 27984
I0324 09:47:02.292835 407313 ExpressionFuzzer.cpp:978] floor 1683 3.18% 27790
I0324 09:47:02.292838 407313 ExpressionFuzzer.cpp:978] divide 1599 3.03% 26578
I0324 09:47:02.292842 407313 ExpressionFuzzer.cpp:978] round 1635 3.09% 26400
I0324 09:47:02.292845 407313 ExpressionFuzzer.cpp:984] ==============================> Bottom 10 by number of rows processed
I0324 09:47:02.292848 407313 ExpressionFuzzer.cpp:986] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I0324 09:47:02.292850 407313 ExpressionFuzzer.cpp:991] empty_approx_set 6 0.01% 4
I0324 09:47:02.292856 407313 ExpressionFuzzer.cpp:991] json_parse 22 0.04% 290
I0324 09:47:02.292860 407313 ExpressionFuzzer.cpp:991] date_diff 141 0.27% 802
I0324 09:47:02.292863 407313 ExpressionFuzzer.cpp:991] week_of_year 160 0.30% 1132
I0324 09:47:02.292866 407313 ExpressionFuzzer.cpp:991] day_of_month 138 0.26% 1298
I0324 09:47:02.292870 407313 ExpressionFuzzer.cpp:991] day_of_week 135 0.26% 1569
I0324 09:47:02.292873 407313 ExpressionFuzzer.cpp:991] bitwise_logical_shift_right 169 0.32% 1605
I0324 09:47:02.292877 407313 ExpressionFuzzer.cpp:991] minute 153 0.29% 1691
I0324 09:47:02.292881 407313 ExpressionFuzzer.cpp:991] doy 159 0.30% 1723
I0324 09:47:02.292884 407313 ExpressionFuzzer.cpp:991] bitwise_and 167 0.32% 1740
I0324 09:47:02.292907 407313 ExpressionFuzzer.cpp:1004] ==============================> All stats sorted by number of times the function was chosen
I0324 09:47:02.292910 407313 ExpressionFuzzer.cpp:1006] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I0324 09:47:02.292913 407313 ExpressionFuzzer.cpp:1010] cast 3162 5.98% 119927
I0324 09:47:02.292917 407313 ExpressionFuzzer.cpp:1010] minus 1712 3.24% 30365
I0324 09:47:02.292920 407313 ExpressionFuzzer.cpp:1010] clamp 1704 3.22% 20896
I0324 09:47:02.292923 407313 ExpressionFuzzer.cpp:1010] abs 1688 3.19% 27995
I0324 09:47:02.292927 407313 ExpressionFuzzer.cpp:1010] multiply 1686 3.19% 30984
I0324 09:47:02.292930 407313 ExpressionFuzzer.cpp:1010] floor 1683 3.18% 27790
I0324 09:47:02.292933 407313 ExpressionFuzzer.cpp:1010] mod 1671 3.16% 27984
I0324 09:47:02.292937 407313 ExpressionFuzzer.cpp:1010] plus 1655 3.13% 28894
I0324 09:47:02.292940 407313 ExpressionFuzzer.cpp:1010] ceiling 1644 3.11% 26107
I0324 09:47:02.292943 407313 ExpressionFuzzer.cpp:1010] round 1635 3.09% 26400
I0324 09:47:02.292946 407313 ExpressionFuzzer.cpp:1010] sign 1626 3.08% 28660
I0324 09:47:02.292950 407313 ExpressionFuzzer.cpp:1010] divide 1599 3.03% 26578
I0324 09:47:02.292953 407313 ExpressionFuzzer.cpp:1010] ceil 1587 3.00% 26296
I0324 09:47:02.292956 407313 ExpressionFuzzer.cpp:1010] negate 1584 3.00% 26137
I0324 09:47:02.292960 407313 ExpressionFuzzer.cpp:1010] date_add 1044 1.98% 6560
I0324 09:47:02.292963 407313 ExpressionFuzzer.cpp:1010] date_trunc 999 1.89% 14468
I0324 09:47:02.292968 407313 ExpressionFuzzer.cpp:1010] codepoint 412 0.78% 4763
I0324 09:47:02.292970 407313 ExpressionFuzzer.cpp:1010] date_parse 410 0.78% 6443
I0324 09:47:02.292973 407313 ExpressionFuzzer.cpp:1010] from_unixtime 390 0.74% 7487
I0324 09:47:02.292977 407313 ExpressionFuzzer.cpp:1010] trim 329 0.62% 4788
I0324 09:47:02.292980 407313 ExpressionFuzzer.cpp:1010] to_hex 326 0.62% 5294
I0324 09:47:02.292984 407313 ExpressionFuzzer.cpp:1010] url_extract_host 318 0.60% 4599
I0324 09:47:02.292987 407313 ExpressionFuzzer.cpp:1010] lower 313 0.59% 4069
I0324 09:47:02.292990 407313 ExpressionFuzzer.cpp:1010] url_encode 313 0.59% 4307
I0324 09:47:02.292994 407313 ExpressionFuzzer.cpp:1010] url_extract_parameter 313 0.59% 3586
I0324 09:47:02.292997 407313 ExpressionFuzzer.cpp:1010] replace 312 0.59% 4299
I0324 09:47:02.293004 407313 ExpressionFuzzer.cpp:1010] format_datetime 309 0.58% 3870
I0324 09:47:02.293007 407313 ExpressionFuzzer.cpp:1010] ltrim 308 0.58% 3705
I0324 09:47:02.293011 407313 ExpressionFuzzer.cpp:1010] regexp_extract 308 0.58% 3340
I0324 09:47:02.293015 407313 ExpressionFuzzer.cpp:1010] upper 308 0.58% 4830
I0324 09:47:02.293017 407313 ExpressionFuzzer.cpp:1010] split_part 307 0.58% 3095
I0324 09:47:02.293021 407313 ExpressionFuzzer.cpp:1010] json_extract_scalar 306 0.58% 4649
I0324 09:47:02.293025 407313 ExpressionFuzzer.cpp:1010] url_extract_path 302 0.57% 3897
I0324 09:47:02.293028 407313 ExpressionFuzzer.cpp:1010] regexp_replace 301 0.57% 2967
I0324 09:47:02.293031 407313 ExpressionFuzzer.cpp:1010] rpad 298 0.56% 2553
I0324 09:47:02.293035 407313 ExpressionFuzzer.cpp:1010] date_format 297 0.56% 3196
I0324 09:47:02.293038 407313 ExpressionFuzzer.cpp:1010] url_decode 296 0.56% 3927
I0324 09:47:02.293041 407313 ExpressionFuzzer.cpp:1010] url_extract_query 295 0.56% 4293
I0324 09:47:02.293045 407313 ExpressionFuzzer.cpp:1010] to_base 295 0.56% 4560
I0324 09:47:02.293048 407313 ExpressionFuzzer.cpp:1010] to_base64 293 0.55% 4538
I0324 09:47:02.293051 407313 ExpressionFuzzer.cpp:1010] url_extract_fragment 292 0.55% 3629
I0324 09:47:02.293056 407313 ExpressionFuzzer.cpp:1010] substr 291 0.55% 4459
I0324 09:47:02.293058 407313 ExpressionFuzzer.cpp:1010] rtrim 290 0.55% 4720
I0324 09:47:02.293062 407313 ExpressionFuzzer.cpp:1010] to_base64url 284 0.54% 5275
I0324 09:47:02.293066 407313 ExpressionFuzzer.cpp:1010] reverse 279 0.53% 4093
I0324 09:47:02.293069 407313 ExpressionFuzzer.cpp:1010] lpad 279 0.53% 3192
I0324 09:47:02.293072 407313 ExpressionFuzzer.cpp:1010] between 272 0.51% 3641
I0324 09:47:02.293076 407313 ExpressionFuzzer.cpp:1010] eq 270 0.51% 7657
I0324 09:47:02.293079 407313 ExpressionFuzzer.cpp:1010] json_format 270 0.51% 2203
I0324 09:47:02.293082 407313 ExpressionFuzzer.cpp:1010] neq 268 0.51% 7408
I0324 09:47:02.293087 407313 ExpressionFuzzer.cpp:1010] is_finite 268 0.51% 11974
I0324 09:47:02.293089 407313 ExpressionFuzzer.cpp:1010] chr 267 0.51% 3952
I0324 09:47:02.293092 407313 ExpressionFuzzer.cpp:1010] url_extract_protocol 259 0.49% 3207
I0324 09:47:02.293097 407313 ExpressionFuzzer.cpp:1010] gt 251 0.47% 6790
I0324 09:47:02.293099 407313 ExpressionFuzzer.cpp:1010] is_json_scalar 251 0.47% 17735
I0324 09:47:02.293103 407313 ExpressionFuzzer.cpp:1010] is_nan 250 0.47% 9807
I0324 09:47:02.293107 407313 ExpressionFuzzer.cpp:1010] lte 244 0.46% 6320
I0324 09:47:02.293109 407313 ExpressionFuzzer.cpp:1010] regexp_like 244 0.46% 6180
I0324 09:47:02.293113 407313 ExpressionFuzzer.cpp:1010] json_array_contains 240 0.45% 5932
I0324 09:47:02.293116 407313 ExpressionFuzzer.cpp:1010] is_infinite 234 0.44% 7913
I0324 09:47:02.293120 407313 ExpressionFuzzer.cpp:1010] not 231 0.44% 6850
I0324 09:47:02.293123 407313 ExpressionFuzzer.cpp:1010] like 231 0.44% 2538
I0324 09:47:02.293128 407313 ExpressionFuzzer.cpp:1010] lt 229 0.43% 6120
I0324 09:47:02.293130 407313 ExpressionFuzzer.cpp:1010] gte 216 0.41% 5376
I0324 09:47:02.293133 407313 ExpressionFuzzer.cpp:1010] distinct_from 214 0.40% 10725
I0324 09:47:02.293138 407313 ExpressionFuzzer.cpp:1010] dow 194 0.37% 2189
I0324 09:47:02.293140 407313 ExpressionFuzzer.cpp:1010] month 188 0.36% 2435
I0324 09:47:02.293144 407313 ExpressionFuzzer.cpp:1010] bitwise_right_shift_arithmetic 186 0.35% 2975
I0324 09:47:02.293148 407313 ExpressionFuzzer.cpp:1010] second 184 0.35% 1962
I0324 09:47:02.293150 407313 ExpressionFuzzer.cpp:1010] bitwise_xor 181 0.34% 3272
I0324 09:47:02.293154 407313 ExpressionFuzzer.cpp:1010] tan 178 0.34% 6406
I0324 09:47:02.293159 407313 ExpressionFuzzer.cpp:1010] json_array_length 177 0.33% 1884
I0324 09:47:02.293164 407313 ExpressionFuzzer.cpp:1010] yow 176 0.33% 2248
I0324 09:47:02.293167 407313 ExpressionFuzzer.cpp:1010] day 175 0.33% 1826
I0324 09:47:02.293171 407313 ExpressionFuzzer.cpp:1010] day_of_year 175 0.33% 2223
I0324 09:47:02.293175 407313 ExpressionFuzzer.cpp:1010] length 174 0.33% 2228
I0324 09:47:02.293184 407313 ExpressionFuzzer.cpp:1010] millisecond 173 0.33% 2213
I0324 09:47:02.293188 407313 ExpressionFuzzer.cpp:1010] bitwise_not 173 0.33% 2452
I0324 09:47:02.293192 407313 ExpressionFuzzer.cpp:1010] xxhash64 173 0.33% 3320
I0324 09:47:02.293196 407313 ExpressionFuzzer.cpp:1010] tanh 171 0.32% 3876
I0324 09:47:02.293200 407313 ExpressionFuzzer.cpp:1010] atan 170 0.32% 5289
I0324 09:47:02.293205 407313 ExpressionFuzzer.cpp:1010] bitwise_shift_left 170 0.32% 1789
I0324 09:47:02.293208 407313 ExpressionFuzzer.cpp:1010] quarter 169 0.32% 2371
I0324 09:47:02.293213 407313 ExpressionFuzzer.cpp:1010] bitwise_logical_shift_right 169 0.32% 1605
I0324 09:47:02.293216 407313 ExpressionFuzzer.cpp:1010] log2 168 0.32% 4047
I0324 09:47:02.293220 407313 ExpressionFuzzer.cpp:1010] week 168 0.32% 2128
I0324 09:47:02.293224 407313 ExpressionFuzzer.cpp:1010] hmac_sha256 168 0.32% 3097
I0324 09:47:02.293228 407313 ExpressionFuzzer.cpp:1010] bitwise_and 167 0.32% 1740
I0324 09:47:02.293233 407313 ExpressionFuzzer.cpp:1010] from_hex 166 0.31% 2401
I0324 09:47:02.293237 407313 ExpressionFuzzer.cpp:1010] crc32 165 0.31% 3179
I0324 09:47:02.293241 407313 ExpressionFuzzer.cpp:1010] log10 164 0.31% 5011
I0324 09:47:02.293246 407313 ExpressionFuzzer.cpp:1010] bitwise_arithmetic_shift_right 164 0.31% 1889
I0324 09:47:02.293251 407313 ExpressionFuzzer.cpp:1010] truncate 163 0.31% 4362
I0324 09:47:02.293254 407313 ExpressionFuzzer.cpp:1010] radians 163 0.31% 5215
I0324 09:47:02.293258 407313 ExpressionFuzzer.cpp:1010] atan2 162 0.31% 4591
I0324 09:47:02.293263 407313 ExpressionFuzzer.cpp:1010] hmac_md5 161 0.30% 2562
I0324 09:47:02.293267 407313 ExpressionFuzzer.cpp:1010] url_extract_port 161 0.30% 2191
I0324 09:47:02.293272 407313 ExpressionFuzzer.cpp:1010] hour 160 0.30% 1826
I0324 09:47:02.293277 407313 ExpressionFuzzer.cpp:1010] week_of_year 160 0.30% 1132
I0324 09:47:02.293280 407313 ExpressionFuzzer.cpp:1010] bitwise_right_shift 159 0.30% 3147
I0324 09:47:02.293283 407313 ExpressionFuzzer.cpp:1010] doy 159 0.30% 1723
I0324 09:47:02.293287 407313 ExpressionFuzzer.cpp:1010] bitwise_left_shift 158 0.30% 2412
I0324 09:47:02.293290 407313 ExpressionFuzzer.cpp:1010] strpos 158 0.30% 2137
I0324 09:47:02.293293 407313 ExpressionFuzzer.cpp:1010] from_base64 158 0.30% 2627
I0324 09:47:02.293298 407313 ExpressionFuzzer.cpp:1010] sqrt 156 0.30% 4498
I0324 09:47:02.293300 407313 ExpressionFuzzer.cpp:1010] sha512 156 0.30% 2560
I0324 09:47:02.293304 407313 ExpressionFuzzer.cpp:1010] hmac_sha512 154 0.29% 2901
I0324 09:47:02.293308 407313 ExpressionFuzzer.cpp:1010] bit_count 154 0.29% 1976
I0324 09:47:02.293311 407313 ExpressionFuzzer.cpp:1010] json_size 154 0.29% 2562
I0324 09:47:02.293314 407313 ExpressionFuzzer.cpp:1010] minute 153 0.29% 1691
I0324 09:47:02.293318 407313 ExpressionFuzzer.cpp:1010] degrees 153 0.29% 4190
I0324 09:47:02.293321 407313 ExpressionFuzzer.cpp:1010] strrpos 153 0.29% 2136
I0324 09:47:02.293324 407313 ExpressionFuzzer.cpp:1010] from_base 152 0.29% 3100
I0324 09:47:02.293329 407313 ExpressionFuzzer.cpp:1010] cbrt 152 0.29% 3496
I0324 09:47:02.293331 407313 ExpressionFuzzer.cpp:1010] to_unixtime 151 0.29% 2124
I0324 09:47:02.293335 407313 ExpressionFuzzer.cpp:1010] year 151 0.29% 2148
I0324 09:47:02.293339 407313 ExpressionFuzzer.cpp:1010] cos 150 0.28% 4503
I0324 09:47:02.293341 407313 ExpressionFuzzer.cpp:1010] acos 150 0.28% 4598
I0324 09:47:02.293345 407313 ExpressionFuzzer.cpp:1010] hmac_sha1 150 0.28% 2629
I0324 09:47:02.293349 407313 ExpressionFuzzer.cpp:1010] exp 150 0.28% 4401
I0324 09:47:02.293352 407313 ExpressionFuzzer.cpp:1010] pow 149 0.28% 3626
I0324 09:47:02.293355 407313 ExpressionFuzzer.cpp:1010] ln 149 0.28% 4971
I0324 09:47:02.293359 407313 ExpressionFuzzer.cpp:1010] sha1 149 0.28% 2364
I0324 09:47:02.293362 407313 ExpressionFuzzer.cpp:1010] power 148 0.28% 3222
I0324 09:47:02.293365 407313 ExpressionFuzzer.cpp:1010] nan 146 0.28% 10147
I0324 09:47:02.293370 407313 ExpressionFuzzer.cpp:1010] sin 146 0.28% 4043
I0324 09:47:02.293372 407313 ExpressionFuzzer.cpp:1010] e 145 0.27% 9540
I0324 09:47:02.293380 407313 ExpressionFuzzer.cpp:1010] sha256 145 0.27% 3352
I0324 09:47:02.293382 407313 ExpressionFuzzer.cpp:1010] to_utf8 145 0.27% 1911
I0324 09:47:02.293385 407313 ExpressionFuzzer.cpp:1010] year_of_week 145 0.27% 1895
I0324 09:47:02.293390 407313 ExpressionFuzzer.cpp:1010] bitwise_or 145 0.27% 1859
I0324 09:47:02.293392 407313 ExpressionFuzzer.cpp:1010] pi 144 0.27% 9837
I0324 09:47:02.293396 407313 ExpressionFuzzer.cpp:1010] asin 144 0.27% 3443
I0324 09:47:02.293399 407313 ExpressionFuzzer.cpp:1010] infinity 143 0.27% 9045
I0324 09:47:02.293403 407313 ExpressionFuzzer.cpp:1010] spooky_hash_v2_32 143 0.27% 1987
I0324 09:47:02.293406 407313 ExpressionFuzzer.cpp:1010] date_diff 141 0.27% 802
I0324 09:47:02.293411 407313 ExpressionFuzzer.cpp:1010] cosh 140 0.26% 3867
I0324 09:47:02.293413 407313 ExpressionFuzzer.cpp:1010] md5 139 0.26% 2458
I0324 09:47:02.293416 407313 ExpressionFuzzer.cpp:1010] day_of_month 138 0.26% 1298
I0324 09:47:02.293421 407313 ExpressionFuzzer.cpp:1010] day_of_week 135 0.26% 1569
I0324 09:47:02.293423 407313 ExpressionFuzzer.cpp:1010] spooky_hash_v2_64 133 0.25% 2582
I0324 09:47:02.293427 407313 ExpressionFuzzer.cpp:1010] json_parse 22 0.04% 290
I0324 09:47:02.293431 407313 ExpressionFuzzer.cpp:1010] empty_approx_set 6 0.01% 4
```

**Comment by czentgr** [comment]
Strange. The pipeline test fails but local test is fine
```
180: [----------] Global test environment tear-down
180: [==========] 577 tests from 56 test suites ran. (1349944 ms total)
180: [  PASSED  ] 577 tests.
183/183 Test #180: velox_exec_test ................................................   Passed  1350.39 sec

100% tests passed, 0 tests failed out of 183

Total Test time (real) = 2890.15 sec
```

**Comment by czentgr** [comment]
Opened new issue for the circleci failure in the fuzzer: https://github.com/facebookincubator/velox/issues/4416

**Comment by czentgr** [comment]
Tests after merging up:

```
180: [==========] 577 tests from 56 test suites ran. (1349944 ms total)
180: [  PASSED  ] 577 tests.
183/183 Test #180: velox_exec_test ................................................   Passed  1350.39 sec

100% tests passed, 0 tests failed out of 183

Total Test time (real) = 2890.15 sec
```

**Comment by czentgr** [comment]
Some of these changes here are also part of PR https://github.com/facebookincubator/velox/pull/4423 (the change from bool to void). Either the other PR is merged first and then this will be a conflict or I remove my changes to avoid conflicts.

File: velox/common/encode/Base64.h [hunk 1]
```diff
@@ -77,14 +77,13 @@ class Base64 {
...
+  static std::string encodeUrl(const folly::IOBuf* data);
+  static std::string encodeUrl(folly::StringPiece text);
+  static void decodeUrl(
       const std::pair<const char*, int32_t>& payload,
       std::string& output);
-  static std::string decode_url(folly::StringPiece text);
+  static std::string decodeUrl(folly::StringPiece text);
 
   /*
    * Take str and writes the encoded string to out.
```
Comment (aditi-pandit) [inline velox/common/encode/Base64.h:89]: Make base64_encode(decode)_string(..) naming consistent also since we are doing this exercise.

**Review by aditi-pandit — COMMENTED** [review]
Thanks for the cleanup.

File: velox/common/encode/Base64.h (hunk 1 above)
Comment (czentgr) [reply velox/common/encode/Base64.h:89]: Turns out that neither `base64_encode_string` nor `base64_decode_string` are implemented. The former has a commented implementation.
I suspect that this was an old implementation which was later replaced by implementing `decodeImpl` and `encodeImpl` and its various wrappers.

I can remove this completely.

**Review by aditi-pandit — APPROVED** [review]
Thanks !

**Review by mbasmanova — APPROVED** [review]
@czentgr Nice. Looks like only function names were modified. Return types seem to be the same. Would be nice to update the PR title.

**Comment by czentgr** [comment]
> @czentgr Nice. Looks like only function names were modified. Return types seem to be the same. Would be nice to update the PR title.

@mbasmanova looks like you already did that :)
Yes, this was the old title before I rebased that included your changes that took care of the same already as mentioned in an earlier comment.
Thanks!

**Pull request closed without merging**